src/Composer/InstalledPackageResolver.php

```diff
@@ -59,6 +59,20 @@ public function resolve(): array
         return $installedPackages;
     }
 
+    public function resolvePackageVersion(string $packageName): ?string
+    {
+        $installedPackages = $this->resolve();
```

Comment:
This may need unit test to extract tagged version; eg

`1.0.0@0beec7b5ea3f0fdbc95d0dd47f3c5bc275da8a33` to become `1.0.0` only